consensus/README.md

```diff
@@ -140,6 +150,244 @@ graph TB
+  * Before applying a block `n`, verify that is is signed by a quorum of `n-1`'s validator set (genesis validator set for block `1`)
+  * By applying each block, the validator set is updated (validators joining or leaving), starting from genesis validator set for any new node
+* With this chain of trust, a synching node systematically detects invalid blocks
+  * No malicious or faulty node could inject an alternative block without making at least `2t+1` validators sign it
+  * The persistence layer is mainly a cache for the block application, so a node won't restart block application from genesis each time it's rebooted
+* When the routine applies `NetworkCurrentHeight`, it signals it so the node could switch to `consensus` mode. Meanwhile, it waits to apply a new downloaded block
+
+_Improvement: After applying a block, the server's signature (the one the block has been downloaded from) is replaced by the current node signature. Meaning that from now on, the current node endorse the block validity (when serving the block to other synchers). It could also serve as means to check the integrity of its own database._
+
+```mermaid
+sequenceDiagram
+  title: Consensus: Apply blocks
+
+  participant Persistence
+  participant SyncRoutine
+  participant BlockApplier
+  participant State
+
+  loop AppliedBlock < NetworkCurrentHeight
+    SyncRoutine->>Persistence:GetDownloadedBlock
+    Persistence-->>SyncRoutine: 
+
+    Note right of Persistence: Or genesis if starting from scratch
+    SyncRoutine->>State:GetValidatorSet
+    State-->>SyncRoutine: 
+
+    SyncRoutine->>SyncRoutine:Verify(block.qc, ValidatorSet)
+
+    SyncRoutine->>BlockApplier:ApplyBlock
+    BlockApplier-->>SyncRoutine: 
+
+    Note left of BlockApplier: Tells if it's valid block and what are the  <br />changes to be made to the state
+    SyncRoutine->>State:Update
+
+    Note left of State: ValidatorSet should be updated here
+    SyncRoutine->>Persistence:MarkBlockAsApplied
+
+    SyncRoutine->>SyncRoutine:Wait(until block downloaded)
+    Note left of BlockApplier: Wait for the next block to apply<br /> if it's not downloaded yet
+
+    SyncRoutine->>SyncRoutine:If all blocks applied:<br />SignalSyncEnd
+    Note right of SyncRoutine: StateSync has finished<br />Switch to consensus mode
+  end
+```
+
+## Consensus Mode
+
+In this mode, the current node is up to date w.r.t. the latest block applied by the network and can start now participating to the consensus process.
+
+This process is driven by:
+* A pace maker, that alerts the consensus flow about key timings in the block production process
+  * `MinBlockTime`: The earliest time a leader node could reap the mempool and start proposing a new block
+  * `RoundTimeout`: How much time a round is allowed to take before calling for another one
+  * Pacemaker uses a local clock and a node is only aware of the global pace through new round events
```

Comment:
Per #953, there is no such concept of time in blockchains so I believe we need to reword some statements here.